twill-core/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -105,4 +114,17 @@ public String getRmSchedulerAddr() {
   public Map<String, Integer> getMaxRetries() {
     return maxRetries;
   }
+
+  /**
+   * Returns the ZK connection string for the Kafka used for log collections,
+   * or {@code null} if log collection is disabled.
+   */
+  @Nullable
+  public String getKafkaZKConnect() {
+    if (!isLogCollectionEnabled()) {
```

Comment:
Same thing of no way that caller could have guessed without looking at the code that there is a check of isLogCollectionEnabled inside getKafkaZKConnect that have inferred meaning when it is returning null.

The compromise I think at least added JavaDoc header for getKafkaZKConnect to explain why it could return null.